### PR TITLE
Feature/tree module

### PR DESCRIPTION
**Pull request by EluciusFTW**
Next up: Trees!

### CODE REVIEW

File: src/spectrecoff/Tree.fs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    | BoldLine -> root.Guide <- TreeGuide.BoldLine
+    root
+
+let attach (nodes: TreeNode list) (node: TreeNode) =
+    nodes 
+    |> List.iter (fun n -> node.AddNode n |> ignore)
+    node
+
+let createNode (content: OutputPayload) (nodes: TreeNode list) = 
+    let node = TreeNode (content |> payloadToRenderable )
```
Comment (DanielMuckelbauer) [inline src/spectrecoff/Tree.fs:45]: `let node = TreeNode(content |> payloadToRenderable)`

File: src/spectrecoff/Tree.fs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    | Collapse -> root.Expanded <- false
+
+    match layout.Guides with
+    | Ascii -> root.Guide <- TreeGuide.Ascii
+    | SingleLine -> root.Guide <- TreeGuide.Line
+    | DoubleLine -> root.Guide <- TreeGuide.DoubleLine
+    | BoldLine -> root.Guide <- TreeGuide.BoldLine
+    root
+
+let attach (nodes: TreeNode list) (node: TreeNode) =
```
Comment (DanielMuckelbauer) [inline src/spectrecoff/Tree.fs:39]: This takes me back to our Fragment Parser :)

File: src/spectrecoff/Tree.fs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    | DoubleLine
+    | BoldLine
+
+type TreeLayout =
+    {  Sizing: SizingBehaviour
+       Guides: GuideStyle }
+
+let defaultTreeLayout: TreeLayout =
+    {  Sizing = Collapse
+       Guides = SingleLine }
```
Comment (DanielMuckelbauer) [inline src/spectrecoff/Tree.fs:20]: Spectre also supports a "Style" method for colors, maybe it could be added as well?

**Review by DanielMuckelbauer — APPROVED** [review]

File: src/spectrecoff-cli/Program.fs [hunk 4]
```diff
@@ -94,6 +94,17 @@ let main argv =
                 .WithDescription("Shows the documentation for the breakdown chart module.")
                 |> ignore)
 
+        config.AddBranch("tree", fun(add: IConfigurator<TreeSettings>) ->
+            add.AddCommand<TreeExample>("example")
+                .WithAlias("e")
+                .WithDescription("Shows examples of tree charts.")
```
[the remark this answers is not in the document]
Comment (EluciusFTW) [reply src/spectrecoff-cli/Program.fs:100]: You got me there ...

File: src/spectrecoff-cli/commands/Tree.fs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    inherit CommandSettings()
+
+type TreeExample() =
+    inherit Command<TreeSettings>()
+    interface ICommandLimiter<TreeSettings>
+
+    override _.Execute(_context, _) =
+
+        let subNode value = 
+            createNode (panel $"" (P value)) []
```
[the remark this answers is not in the document]
Comment (EluciusFTW) [reply src/spectrecoff-cli/commands/Tree.fs:17]: It's called the EmptyInterpolation. Advanced stuff ...

[the remark this answers is not in the document]
Comment (EluciusFTW) [reply src/spectrecoff-cli/commands/Tree.fs]: Fixed.

[the remark this answers is not in the document]
Comment (DanielMuckelbauer) [reply src/spectrecoff-cli/commands/Tree.fs]: Hihi

**Review by DanielMuckelbauer — COMMENTED** [review]
Now that I think of it, you should probably also adjust the progress command to reflect the inclusion of the table and also replace the screenshot of it.

File: src/spectrecoff-cli/Program.fs (hunk 4 above)
[the remark this answers is not in the document]
Comment (EluciusFTW) [reply src/spectrecoff-cli/Program.fs]: Fixed.

File: src/spectrecoff/Tree.fs (hunk 2 above)
Comment (EluciusFTW) [reply src/spectrecoff/Tree.fs:39]: Let's think about reviving that one ... when we're done here :p

File: src/spectrecoff/Tree.fs (hunk 3 above)
Comment (EluciusFTW) [reply src/spectrecoff/Tree.fs:20]: I'll look at that in a follow-up, when I add the documentation as well.

**Comment by EluciusFTW** [comment]
@DanielMuckelbauer Updated the progress like suggested!

File: src/spectrecoff/Tree.fs (hunk 1 above)
Comment (EluciusFTW) [reply src/spectrecoff/Tree.fs]: Refactored the code, no more braces ;)

**Pull request merged**